crates/uv-distribution/src/workspace.rs

```diff
@@ -824,6 +808,52 @@ fn is_excluded_from_workspace(
     Ok(false)
 }
 
+/// An iterator over a [`Pattern`] in a workspace.
+///
+/// Iterates from the workspace root, returning paths that match the pattern.
+#[derive(Debug)]
+struct WorkspaceGlobIter<'a> {
+    root: &'a Path,
+    glob: &'a Pattern,
+    iter: walkdir::IntoIter,
+}
+
+impl<'a> WorkspaceGlobIter<'a> {
+    fn new(root: &'a Path, glob: &'a Pattern) -> Self {
+        Self {
+            root,
+            glob,
+            iter: walkdir::WalkDir::new(root).into_iter(),
+        }
+    }
+}
+
+impl<'a> Iterator for WorkspaceGlobIter<'a> {
+    type Item = Result<PathBuf, walkdir::Error>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            let entry = match self.iter.next()? {
+                Ok(entry) => entry,
+                Err(err) => return Some(Err(err)),
+            };
+            if entry.file_type().is_dir() {
+                if self.glob.matches_path(entry.path())
+                    || entry
+                        .path()
+                        .strip_prefix(self.root)
+                        .is_ok_and(|path| self.glob.matches_path(path))
```

Comment:
I think you _only_ want the second condition here. That is, you always want to match the paths with globs _relative_ to the project root. And every path returned by `walkdir` is, I believe, guaranteed to always have the [root path prefix](https://docs.rs/walkdir/latest/walkdir/struct.DirEntry.html#method.path).

Stated differently, I think the first `self.glob.matches_path(entry.path())` should be removed here.